src/cogent3/app/__init__.py

```diff
@@ -52,36 +56,67 @@ def _make_types(app) -> dict:
     return _types
 
 
-def available_apps(name_filter: str | None = None) -> Table:
+# private global to hold an ExtensionManager instance
+__apps = None
+
+
+def apps(force: bool = False) -> stevedore.ExtensionManager:
+    """
+    Lazy load a stevedore ExtensionManager to collect apps.
+
+    Parameters
+    ----------
+    force : bool, optional
+        If set to True, forces the re-creation of the ExtensionManager,
+        even if it already exists. Default is False.
+
+    Returns
+    -------
+    ExtensionManager
+        The ExtensionManager instance that collects apps.
+    """
+    global __apps
+    if force:
+        importlib.invalidate_caches()
+        importlib.reload(stevedore)
+        stevedore.extension.ExtensionManager.ENTRY_POINT_CACHE.clear()
+    if not __apps or force:
+        __apps = stevedore.ExtensionManager(
+            namespace="cogent3.app", invoke_on_load=False
+        )
+    return __apps
+
+
+def available_apps(name_filter: str | None = None, force: bool = False) -> Table:
     """returns Table listing the available apps
 
     Parameters
     ----------
     name_filter
         include apps whose name includes name_filter
+    force : bool, optional
```

Comment:
I think leave all type hint info for the function signature, otherwise chance of out-of-sync